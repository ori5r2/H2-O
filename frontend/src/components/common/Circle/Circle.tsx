import type { SVGProps } from 'react';
import { ReactComponent as CircleBorder } from '@/assets/shape/circle-border.svg';
import { ReactComponent as CircleFill } from '@/assets/shape/circle-fill.svg';

interface Props extends SVGProps<SVGSVGElement> {
  type: 'fill' | 'border';
  color?: string;
  size?: number;
}

function Circle({ type, color, size, ...restProps }: Props) {
  return type === 'fill' ? (
    <CircleFill fill={color} width={size ? size : 46} height={size ? size : 46} {...restProps} />
  ) : (
    <CircleBorder stroke={color} width={size ? size : 58} height={size ? size : 58} {...restProps} />
  );
}

export default Circle;
