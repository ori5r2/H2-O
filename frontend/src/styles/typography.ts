export const TYPOGRAPHY = {
  HeadEnBold26: {
    fontFamily: 'Hyundai Sans Head',
    fontSize: '26px',
    fontWeight: 700,
    lineHeight: '36px',
    letterSpacing: '-0.01em',
  },
  HeadENBold24: {
    fontFamily: 'Hyundai Sans Head',
    fontSize: '24px',
    fontWeight: 700,
    lineHeight: '32px',
    letterSpacing: '-0.01em',
  },
  HeadENBold22: {
    fontFamily: 'Hyundai Sans Head',
    fontSize: '22px',
    fontWeight: 700,
    lineHeight: '28px',
    letterSpacing: '-0.01em',
  },
  HeadENMedium20: {
    fontFamily: 'Hyundai Sans Head',
    fontSize: '20px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.01em',
  },
  HeadENBold10: {
    fontFamily: 'Hyundai Sans Head',
    fontSize: '10px',
    fontWeight: 700,
    lineHeight: '12px',
    letterSpacing: '-0.01em',
  },

  TextENMedium18: {
    fontFamily: 'Hyundai Sans Text',
    fontSize: '18px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.01em',
  },
  TextENMedium16: {
    fontFamily: 'Hyundai Sans Text',
    fontSize: '16px',
    fontWeight: 500,
    lineHeight: '24px',
    letterSpacing: '-0.01em',
  },
  TextENMedium14: {
    fontFamily: 'Hyundai Sans Text',
    fontSize: '14px',
    fontWeight: 500,
    lineHeight: '22px',
    letterSpacing: '-0.01em',
  },
  CaptionENMedium14: {
    fontFamily: 'Hyundai Sans Text',
    fontSize: '14px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.01em',
  },
  CaptionENMedium12: {
    fontFamily: 'Hyundai Sans Text',
    fontSize: '12px',
    fontWeight: 500,
    lineHeight: '24px',
    letterSpacing: '-0.01em',
  },

  HeadKRBold32: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '32px',
    fontWeight: 700,
    lineHeight: '36px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold26: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '26px',
    fontWeight: 700,
    lineHeight: '36px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold24: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '24px',
    fontWeight: 700,
    lineHeight: '32px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold22: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '22px',
    fontWeight: 700,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold20: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '20px',
    fontWeight: 700,
    lineHeight: '24px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold18: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '18px',
    fontWeight: 700,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRBold12: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '12px',
    fontWeight: 700,
    lineHeight: '18px',
  },

  TextKRBold12: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '12px',
    fontWeight: 700,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },

  HeadKRMedium26: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '26px',
    fontWeight: 500,
    lineHeight: '36px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium24: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '24px',
    fontWeight: 500,
    lineHeight: '32px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium22: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '22px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium20: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '20px',
    fontWeight: 500,
    lineHeight: '24px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium18: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '18px',
    fontWeight: 500,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium16: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '16px',
    fontWeight: 500,
    lineHeight: '24px',
    letterSpacing: '-0.03em',
  },
  HeadKRMedium14: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '14px',
    fontWeight: 500,
    lineHeight: '22px',
    letterSpacing: '-0.03em',
  },

  TextKRMedium18: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '18px',
    fontWeight: 500,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRMedium16: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '16px',
    fontWeight: 500,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRMedium14: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '14px',
    fontWeight: 500,
    lineHeight: '22px',
    letterSpacing: '-0.03em',
  },
  TextKRMedium12: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '12px',
    fontWeight: 500,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRMedium10: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '10px',
    fontWeight: 500,
    lineHeight: '16px',
    letterSpacing: '-0.03em',
  },

  HeadKRRegular26: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '26px',
    fontWeight: 400,
    lineHeight: '36px',
    letterSpacing: '-0.03em',
  },
  HeadKRRegular24: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '24px',
    fontWeight: 400,
    lineHeight: '32px',
    letterSpacing: '-0.03em',
  },
  HeadKRRegular22: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '22px',
    fontWeight: 400,
    lineHeight: '28px',
    letterSpacing: '-0.03em',
  },
  HeadKRRegular20: {
    fontFamily: 'Hyundai Sans Head KR',
    fontSize: '20px',
    fontWeight: 400,
    lineHeight: '24px',
    letterSpacing: '-0.03em',
  },

  TextKRRegular18: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '18px',
    fontWeight: 400,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRRegular16: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '16px',
    fontWeight: 400,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRRegular14: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '14px',
    fontWeight: 400,
    lineHeight: '22px',
    letterSpacing: '-0.03em',
  },
  TextKRRegular12: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '12px',
    fontWeight: 400,
    lineHeight: '18px',
    letterSpacing: '-0.03em',
  },
  TextKRRegular10: {
    fontFamily: 'Hyundai Sans Text KR',
    fontSize: '10px',
    fontWeight: 400,
    lineHeight: '16px',
    letterSpacing: '-0.03em',
  },
} as const;

export type TypographyType = typeof TYPOGRAPHY;
export type TypographVariant = keyof TypographyType;
