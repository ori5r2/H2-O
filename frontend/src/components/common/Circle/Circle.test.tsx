import Circle from './Circle';
import { render, screen } from '@/tests/test-util';
import { COLORS } from '@/styles/colors';

describe('Circle 컴포넌트 테스트', () => {
  it('Circle의 type 속성 fill이 정상적으로 렌더링 된다.', () => {
    render(<Circle data-testid='circle-fill' type='fill' />);

    expect(screen.getByTestId('circle-fill')).toBeInTheDocument();
  });

  it('Cirlce의 type 속성 border가 정상적으로 렌더링 된다.', () => {
    render(<Circle data-testid='circle-border' type='border' />);

    expect(screen.getByTestId('circle-border')).toBeInTheDocument();
  });

  it('Circle의 type 속성이 fill인 컴포넌트는 color 속성을 통해 내부 색상을 조절할 수 있다.', () => {
    render(<Circle data-testid='circle-fill' type='fill' color={COLORS.activeBlue} />);

    expect(screen.getByTestId('circle-fill')).toHaveAttribute('fill', COLORS.activeBlue);
  });

  it('Circle의 type 속성이 border인 컴포넌트는 color 속성을 통해 테두리 색상을 조절할 수 있다.', () => {
    render(<Circle data-testid='circle-border' type='border' color={COLORS.primary500} />);

    expect(screen.getByTestId('circle-border')).toHaveAttribute('stroke', COLORS.primary500);
  });

  it('Circle의 type 속성이 fill인 컴포넌트의 기본 사이즈는 58이다.', () => {
    render(<Circle data-testid='circle-fill' type='fill' />);

    expect(screen.getByTestId('circle-fill')).toHaveAttribute('width', '46');
    expect(screen.getByTestId('circle-fill')).toHaveAttribute('height', '46');
  });

  it('Circle의 type 속성이 border인 컴포넌트의 기본 사이즈는 58이다', () => {
    render(<Circle data-testid='circle-border' type='border' />);

    expect(screen.getByTestId('circle-border')).toHaveAttribute('width', '58');
    expect(screen.getByTestId('circle-border')).toHaveAttribute('height', '58');
  });

  it('Circle의 size를 통해 크기를 조절할 수 있다.', () => {
    render(<Circle data-testid='circle-fill' type='fill' size={20} />);

    expect(screen.getByTestId('circle-fill')).toHaveAttribute('width', '20');
    expect(screen.getByTestId('circle-fill')).toHaveAttribute('height', '20');
  });

  describe('Circle 컴포넌트 스냅샷 테스트', () => {
    it('Circle의 type 속성이 fill인 컴포넌트 스냅샷 테스트', () => {
      render(<Circle data-testid='circle-fill' type='fill' />);

      expect(screen.getByTestId('circle-fill')).toMatchSnapshot();
    });

    it('Circle의 type 속성이 border인 컴포넌트 스냅샷 테스트', () => {
      render(<Circle data-testid='circle-border' type='border' />);

      expect(screen.getByTestId('circle-border')).toMatchSnapshot();
    });
  });
});
