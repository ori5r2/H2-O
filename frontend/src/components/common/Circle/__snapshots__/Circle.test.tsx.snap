// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Circle 컴포넌트 테스트 > Circle 컴포넌트 스냅샷 테스트 > Circle의 type 속성이 border인 컴포넌트 스냅샷 테스트 1`] = `
<svg
  data-testid="circle-border"
  fill="none"
  height="58"
  viewBox="0 0 58 58"
  width="58"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    filter="url(#filter0_i_2836_21)"
    id="Ellipse 657"
  >
    <circle
      cx="29"
      cy="29"
      r="28.5"
      stroke="inherit"
    />
  </g>
  <defs>
    <filter
      color-interpolation-filters="sRGB"
      filterUnits="userSpaceOnUse"
      height="58"
      id="filter0_i_2836_21"
      width="58"
      x="0"
      y="0"
    >
      <feflood
        flood-opacity="0"
        result="BackgroundImageFix"
      />
      <feblend
        in="SourceGraphic"
        in2="BackgroundImageFix"
        mode="normal"
        result="shape"
      />
      <fecolormatrix
        in="SourceAlpha"
        result="hardAlpha"
        type="matrix"
        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
      />
      <feoffset />
      <fegaussianblur
        stdDeviation="2"
      />
      <fecomposite
        in2="hardAlpha"
        k2="-1"
        k3="1"
        operator="arithmetic"
      />
      <fecolormatrix
        type="matrix"
        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.1 0"
      />
      <feblend
        in2="shape"
        mode="normal"
        result="effect1_innerShadow_2836_21"
      />
    </filter>
  </defs>
</svg>
`;

exports[`Circle 컴포넌트 테스트 > Circle 컴포넌트 스냅샷 테스트 > Circle의 type 속성이 fill인 컴포넌트 스냅샷 테스트 1`] = `
<svg
  data-testid="circle-fill"
  height="46"
  viewBox="0 0 46 46"
  width="46"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    id="Ellipse 651"
  >
    <g
      filter="url(#filter0_i_2836_22)"
      id="Ellipse 650"
    >
      <circle
        cx="23"
        cy="23"
        fill="inherit"
        r="23"
      />
    </g>
  </g>
  <defs>
    <filter
      color-interpolation-filters="sRGB"
      filterUnits="userSpaceOnUse"
      height="46"
      id="filter0_i_2836_22"
      width="46"
      x="0"
      y="0"
    >
      <feflood
        flood-opacity="0"
        result="BackgroundImageFix"
      />
      <feblend
        in="SourceGraphic"
        in2="BackgroundImageFix"
        mode="normal"
        result="shape"
      />
      <fecolormatrix
        in="SourceAlpha"
        result="hardAlpha"
        type="matrix"
        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
      />
      <feoffset />
      <fegaussianblur
        stdDeviation="2"
      />
      <fecomposite
        in2="hardAlpha"
        k2="-1"
        k3="1"
        operator="arithmetic"
      />
      <fecolormatrix
        type="matrix"
        values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.1 0"
      />
      <feblend
        in2="shape"
        mode="normal"
        result="effect1_innerShadow_2836_22"
      />
    </filter>
  </defs>
</svg>
`;
